File: backoffice/src/product/domain/exceptions/ProductNotFoundException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain        import DomainException
from ..valueObjects.ProductId import ProductId

"""
 *
 * Class 
 *
"""

class ProductNotFoundException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __PRODUCT_NOT_FOUND = 123

    """
     *
     * Methods 
     *
    """

    def __init__( self, id : ProductId ) -> None:
        super().__init__( 
            self.__PRODUCT_NOT_FOUND,
            'The product {} has not been found'.format( id.getValue() )
        )

File: preparation/src/product/domain/ProductNotFoundException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class ProductNotFoundException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int PRODUCT_NOT_FOUND = 123;

        /* 
         * 
         * Methods
         *
        */

        public ProductNotFoundException( ProductId id ) : base( 
            PRODUCT_NOT_FOUND,
            string.Format( "The product {0} has not been found", id.getValue() )
        ) {
        }

    }

}